libnetwork/osl/namespace_linux.go

```diff
@@ -39,13 +38,8 @@ func init() {
 }
 
 var (
-	once             sync.Once
-	garbagePathMap   = make(map[string]bool)
-	gpmLock          sync.Mutex
-	gpmWg            sync.WaitGroup
-	gpmCleanupPeriod = 60 * time.Second
-	gpmChan          = make(chan chan struct{})
-	netnsBasePath    = filepath.Join(defaultPrefix, "netns")
+	once          sync.Once
+	netnsBasePath = filepath.Join(defaultPrefix, "netns")
```

Comment:
Not related to this PR probably, but I was curious what this `once` was used for (we should probably rename it to describe what it's for); but looking at that `once`, the logic seems ... iffy .. it's used to call `createBasePath` (which is "ok"), but `createBasePath` depends on `basepath()`, which returns `netnsBasePath`, which can be set through an exported function `SetBasePath()`;

- the `basepath()` function for sure looks redundant (but that's a nit)
- `SetBasePath()` has no synchronisation
- It also means that `SetBasePath` can be called _after_ `createBasePath` ran, and it would not run again (if `netnsBasePath` was modified



https://github.com/moby/moby/blob/577a7958bb6e362bc5d8f219d95ccc5974e9281f/libnetwork/osl/namespace_linux.go#L51-L68